### PR TITLE
Fix #2065 Leverage async entity release

### LINKED ISSUE #2065: Make use of EntityClientEndpoint.release
With the addition of `EntityClientEndpoint.release()` through terracotta-oss/terracotta-apis#257 we can now close entities without being fully blocked in case the server is gone.

This should allow to clean up the code path for removing a `Cache` or closing a `CacheManager`.

### PR DESCRIPTION
**Pull request by ljacomet**

### CODE REVIEW

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java [hunk 1]
```diff
@@ -246,18 +246,14 @@ private void silentDestroy() {
...
-     */
     entityFactory = null;
     inMaintenance = false;
 
+    for (ClusterTierClientEntity clusterTierClientEntity : clusterTierEntities.values()) {
+      clusterTierClientEntity.close();
+    }
     clusterTierEntities.clear();
-    entity = null;
+    entity.close();
```
Comment (henri-tremblay) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java]: No need to nullate anymore? To free memory

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/SimpleClusterTierManagerClientEntity.java [hunk 2]
```diff
@@ -104,7 +105,20 @@ public void setClientId(UUID clientId) {
...
+      Thread.currentThread().interrupt();
+    } catch (ExecutionException e) {
+      Throwable cause = e.getCause();
+      if (cause instanceof RuntimeException) {
+        throw (RuntimeException) cause;
+      } else {
+        throw new RuntimeException(cause);
+      }
+    } catch (TimeoutException e) {
+      LOGGER.debug("Received TimeoutException trying to close", e);
```
Comment (henri-tremblay) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/SimpleClusterTierManagerClientEntity.java]: Isn't debug a bit low?

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/SimpleClusterTierManagerClientEntity.java [hunk 3]
```diff
@@ -104,7 +105,20 @@ public void setClientId(UUID clientId) {
 
   @Override
   public void close() {
-    endpoint.close();
+    try {
+      endpoint.release().get(2, TimeUnit.SECONDS);
```
Comment (henri-tremblay) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/SimpleClusterTierManagerClientEntity.java]: 2 seconds seems a bit quick. It doesn't matter that much if it fails?

File: clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java [hunk 4]
```diff
@@ -142,7 +143,20 @@ void fireDisconnectionEvent() {
   public void close() {
     reconnectListener = null;
     disconnectionListener = null;
-    endpoint.close();
+    try {
+      endpoint.release().get(2, TimeUnit.SECONDS);
```
Comment (henri-tremblay) [inline clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java]: Same comments as above

File: clustered/integration-test/src/test/java/org/ehcache/clustered/management/AbstractClusteringManagementTest.java [hunk 5]
```diff
@@ -41,14 +39,15 @@
 import org.terracotta.management.model.message.Message;
 import org.terracotta.management.model.notification.ContextualNotification;
 import org.terracotta.management.model.stats.ContextualStatistics;
-import org.terracotta.testing.rules.BasicExternalCluster;
 import org.terracotta.testing.rules.Cluster;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (henri-tremblay) [inline clustered/integration-test/src/test/java/org/ehcache/clustered/management/AbstractClusteringManagementTest.java]: Probably unused imports

**Comment by ljacomet** [comment]
Blocked by passthrough issues - closing for now

**Pull request closed without merging**